metals/src/main/scala/scala/meta/internal/metals/Trace.scala

```diff
@@ -14,25 +14,31 @@ import dev.dirs.ProjectDirectories
 /**
  * Manages JSON-RPC tracing of incoming/outgoing messages via BSP and LSP.
  */
-object GlobalTrace {
+object Trace {
 
-  /**
-   * Returns a printer to trace JSON messages if the user opts into it.
-   */
-  def setup(protocolName: String): PrintWriter = {
+  def protocolTracePath(
+      protocolName: String,
+      workspace: AbsolutePath = PathIO.workingDirectory
+  ): AbsolutePath = {
+    val traceFilename = s"${protocolName.toLowerCase}.trace.json"
+    globalDirectory.resolve(".metals").resolve(traceFilename)
+  }
+
+  def setup(
+      protocolName: String,
+      workspace: AbsolutePath = PathIO.workingDirectory
+  ): Option[PrintWriter] = {
     MetalsLogger.redirectSystemOut(globalLog)
-    setupTracePrinter(protocolName)
+    setupTracePrinter(protocolName, workspace)
   }
```

Comment:
Apparently, this method is needed in order to start the server. Otherwise, it hangs out at accepting socket connection. I don't know why redirecting output stream is so important, but if it works I don't wanna change it :P

Reply:
Maybe we could redirect it to `.metals/metals.log` ?

Reply:
`.metals/metals.log` seems like a good replacement for `global.log` 👍